cpp/dspace/doi.cc

```diff
@@ -0,0 +1,100 @@
+#include <iostream>
+#include <regex>
+#include <set>
+#include <unordered_map>
+#include <cstdlib>
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "DbRow.h"
+#include "Downloader.h"
+#include "EmailSender.h"
+#include "HttpHeader.h"
+#include "JSON.h"
+#include "StlHelpers.h"
+#include "StringUtil.h"
+#include "VuFind.h"
+#include "util.h"
+
+using namespace std;
+
+const std::string DSPACE_SERVER_URL("https://publikationen.uni-tuebingen.de/rest/items/");
+const std::string DOI_KEY("dc.relation.uri");
+const std::string PUBLIKATION_TITLE_KEY("dc.title");
+unsigned int INSTANCE_COUNT(15000);
+const std::vector<std::string> NOTIFICATION_EMAIL_ADRESSES = { "andrii.lysohor@uni-tuebingen.de" };
+
+void SendNotificationsForDOI(const std::string email_subject, const std::string email_message) {
```

Comment:
Whenever we pass a const std::string to a function, it should always be passed as a reference for performance reasons, e.g. `const std::string &email_subject`, `const std::string &email_message`, and so on (the **&** is very important).